### ISSUE TITLE
♿Checkbox AVT1 - August 2018♿

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#1330]
## Environment
Chrome Version 70.0.3538.67 (Official Build) (64-bit)
macOS Mojave 10/Windows 10

## Violations:
Minimum contrast of 3.36 with its background is not sufficient to meet WCAG AA requirements for text of size 14px and weight of 400 on the disabled checkbox element.

## How to reproduce
1. [Download the Dynamic Accessibility Plugin tool here](https://ibm.ent.box.com/notes/335005856253?s=mhpgjlgss354mcidsv6rvpuu0bxlmfm4)
2. Run the extension using the August 2018 ruleset

**Comment by joshblack** [comment]
Do we need to meet the contrast requirement for a disabled state? I always thought that we didn't. cc @aagonzales 

Would it be possible to include a screenshot of the node that causes the violation?

**Comment by dakahn** [comment]
The rule states _The visual presentation of text and images of text has a contrast ratio of at least 4.5:1, with a 3:1 ratio for large-scale text_. with an incidental exception for inactive interface components. So the design follows the letter of the law.

There is an issue where if the user is in High Contrast mode the only visual indication that the text is disabled is a cursor change on hover, but we might be fine with that as well. Primarily, the text disappears in lower contrast or a bright environment. 

If we are cool with it this ticket can just be used to document a false positive for help with future audits.

**Comment by dakahn** [comment]
[Here's a really interesting discussion that challenges the general (re: our) interpretation of WCAG guidelines for the incidental contrast exception for an "inactive element"
](https://github.com/w3c/wcag21/issues/805)
TLDR: "If an element is communicating its disabled, then users should have the ability to read the information without struggling to read within low contrast. With this constraint, designers would have to reconsider how to communicate disabled states."

"Disabled fields are visible to some, so they should be visible to all."

**Comment by shixiedesign** [comment]
Jeoff: Consider solutions to accessibility that's not contrast. Maybe on hover there can be a tooltip/message saying "this is disabled" etc

**Comment by aagonzales** [comment]
@dakahn is it all component disabled states or is this only related specifically to checkbox?

**Comment by dakahn** [comment]
@aagonzales This violation is specifically for checkbox.

**Comment by snidersd** [comment]
To correct this DAP violation you can add aria-disabled="true" to the label for the disabled control.

**Comment by snidersd** [comment]
Although the recommended fix will remove the violation it turns out that there is a DAP [GitHub issue 1184](https://github.ibm.com/IBMa/Tools-Rules-HTML/pull/1186) that is causing the violation in error and it should be fixed in the next DAP release expected later this month.  This should be re-tested with the new ruleset.

**Comment by dakahn** [comment]
I'll close this for now and we'll test it on the next DAP release. Thanks all!
